### PR TITLE
Add usage of Jupyter magic command for loggers

### PR DESCRIPTION
**Pull request by manangoel99**
## What does this PR do?

This PR adds usage of a jupyter magic command for wandb

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3418,3 +3418,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ## [0.2.x] - 2019-07-09
 
 ## [0.1.x] - 2019-06-DD
+
+### Fixed
+
+- Rearranged documentation for 3rd party loggers.
+
+### Added
+- `%%wandb` magic command to the WandbLogger usage
```
Comment (ananthsub) [inline CHANGELOG.md:3427]: 1. docs changes typically don't require changelog entries
2. this is the oldest section in the changelog - newer updates should be added at the top

File: docs/source/api_references.rst [hunk 2]
```diff
@@ -86,13 +86,14 @@ Loggers API
     :nosignatures:
 
     base
+    wandb
```
Comment (ananthsub) [inline docs/source/api_references.rst:89]: this list is alphabetized. why is `wandb` being moved up?

File: docs/source/common/loggers.rst [hunk 3]
```diff
@@ -28,6 +28,56 @@ the path.
         img = ...
         log_image(img, self.trainer.log_dir)
 
+Weights and Biases
```
Comment (ananthsub) [inline docs/source/common/loggers.rst:31]: same question here

**Comment by manangoel99** [comment]
I've made the suggested changes and pulled the commits from master to this branch to get it up-to-date

File: docs/source/extensions/logging.rst [hunk 4]
```diff
@@ -27,12 +27,12 @@ The following are loggers we support:
     :nosignatures:
     :template: classtemplate.rst
 
+    WandbLogger
```
Comment (krshrimali) [inline docs/source/extensions/logging.rst:30]: As @ananthsub mentioned, can you please keep the list alphabetized?

File: docs/source/extensions/logging.rst [hunk 5]
```diff
@@ -27,12 +27,12 @@ The following are loggers we support:
...
     CometLogger
     CSVLogger
     MLFlowLogger
     NeptuneLogger
     TensorBoardLogger
```
Comment (krshrimali) [inline docs/source/extensions/logging.rst:35]:
```suggestion
    CometLogger
    CSVLogger
    MLFlowLogger
    NeptuneLogger
    TensorBoardLogger
    WandbLogger
```

**Review by krshrimali — CHANGES_REQUESTED** [review]
Hi, @manangoel99 - Thanks for the PR! Left a minor comment, but also curious: from the description of this PR: `This PR rearranges the logger documentation and adds usage of jupyter magic command for wandb`.

Can you please point me to the change (line of code) which adds usage of jupyter magic command for wandb in this PR?

I'm also concerned about re-arranging, unless and until it was discussed in an issue before or has a reason that everyone agrees with (if it was discussed before, please feel free to point me to the discussion :)). cc: @kaushikb11 @awaelchli @Borda

File: docs/source/common/loggers.rst [hunk 6]
```diff
@@ -28,6 +28,56 @@ the path.
...
+            # Option 1
+            self.logger.experiment.log({"generated_images": [wandb.Image(some_img, caption="...")]})
+            # Option 2 for specifically logging images
+            self.logger.log_image(key="generated_images", images=[some_img])
+
+To visualize wandb in a jupyter notebook environment, run the following command in a jupyter cell:
+
+.. code-block:: bash
+
+    %%wandb
```
Comment (rohitgr7) [inline docs/source/common/loggers.rst:72]: as @krshrimali mentioned, the addition of this part is fine. Other seems unimportant. You can extend this one too if there's anything more to add here.

**Comment by manangoel99** [comment]
I've put everything back in the alphabetical order and added usage of the `%%wandb` magic command.

File: docs/source/common/loggers.rst [hunk 7]
```diff
@@ -217,6 +217,12 @@ The :class:`~pytorch_lightning.loggers.WandbLogger` is available anywhere except
             # Option 2 for specifically logging images
             self.logger.log_image(key="generated_images", images=[some_img])
 
+To visualize using wandb in a jupyter notebook environment use the following magic line command
```
Comment (rohitgr7) [inline docs/source/common/loggers.rst:220]:
```suggestion
To visualize using wandb in a jupyter notebook environment use the following magic line command:
```

File: docs/source/common/loggers.rst [hunk 8]
```diff
@@ -217,6 +217,12 @@ The :class:`~pytorch_lightning.loggers.WandbLogger` is available anywhere except
...
+
+.. code-block:: shell
+
+    %%wandb
```
Comment (rohitgr7) [inline docs/source/common/loggers.rst:224]: does it take any arguments here?

File: docs/source/common/loggers.rst (hunk 7 above)
Comment (rohitgr7) [inline docs/source/common/loggers.rst:220]: can you extend this example?
visualize after training or during training?

or maybe link this too: https://docs.wandb.ai/guides/track/jupyter#additional-jupyter-features-in-w-and-b

**Comment by krshrimali** [comment]
Thanks, @manangoel99 for updating this PR. Appreciate the contribution. Just FYI (apart from @rohitgr7's comments) - I updated the description and title of this PR, feel free to make further edits if any changes required. :)

File: docs/source/common/loggers.rst [hunk 9]
```diff
@@ -217,6 +217,30 @@ The :class:`~pytorch_lightning.loggers.WandbLogger` is available anywhere except
...
+    # Display a single run
+    %wandb USERNAME/PROJECT/runs/RUN_ID
+    # Display a sweep
+    %wandb USERNAME/PROJECT/sweeps/SWEEP_ID
+    # Display a report
+    %wandb USERNAME/PROJECT/reports/REPORT_ID
+    # Specify the height of embedded iframe
+    %wandb USERNAME/PROJECT -h 2048
+
+More information is available `here https://docs.wandb.ai/guides/track/jupyter` __.
```
Comment (rohitgr7) [inline docs/source/common/loggers.rst:242]:
```suggestion
More information is available `here https://docs.wandb.ai/guides/track/jupyter`__.
```

**Review by rohitgr7 — APPROVED** [review]

**Review by akihironitta — COMMENTED** [review]
@manangoel99 Could we document a magic command for tensorboard as well in this PR?

**Comment by krshrimali** [comment]
> @manangoel99 Could we document a magic command for tensorboard as well in this PR?

That would be a good addition, thanks @akihironitta for suggesting this. @manangoel99 - if you feel comfortable doing it in this PR, that would be great, but if you want to do it in another PR, that's totally fine as well. Just let us know :)

**Comment by rohitgr7** [comment]
lets add it here

**Comment by manangoel99** [comment]
@krshrimali @rohitgr7 Thanks for adding this to the PR. I actually found the Tensorboard instructions elsewhere so I skipped those initially.
https://pytorch-lightning.readthedocs.io/en/latest/extensions/logging.html

**Comment by rohitgr7** [comment]
yep! I noticed. But we will probably remove tensorboard as default in the future so the existing one will be updated then.

File: docs/source/common/loggers.rst [hunk 10]
```diff
@@ -175,6 +175,19 @@ The :class:`~pytorch_lightning.loggers.TensorBoardLogger` is available anywhere
             some_img = fake_image()
             self.logger.experiment.add_image("generated_images", some_img, 0)
 
+To see your logs in terminal, run the following command:
```
Comment (awaelchli) [inline docs/source/common/loggers.rst:178]: but you won't see the logs in the terminal :( 

```suggestion
To see your logs, run the following command in the terminal:
```

File: docs/source/common/loggers.rst [hunk 11]
```diff
@@ -217,6 +230,31 @@ The :class:`~pytorch_lightning.loggers.WandbLogger` is available anywhere except
             # Option 2 for specifically logging images
             self.logger.log_image(key="generated_images", images=[some_img])
 
+To visualize using wandb in a jupyter notebook environment use the following magic line command:
```
Comment (awaelchli) [inline docs/source/common/loggers.rst:233]: this is all nice and useful, but we need to be careful not to replicate wandb documentation into our own docs. just as a note for the future.

**Review by awaelchli — APPROVED** [review]

**Review by krshrimali — APPROVED** [review]
LGTM, thanks @manangoel99 for your contribution.

**Comment by carmocca** [comment]
@manangoel99 There are some docs failures:

`/home/runner/work/pytorch-lightning/pytorch-lightning/docs/source/common/loggers.rst:: ERROR: Anonymous hyperlink mismatch: 1 references but 0 targets.`

cc @rohitgr7 (if you know what's the cause)

File: docs/source/common/loggers.rst [hunk 12]
```diff
@@ -217,6 +230,31 @@ The :class:`~pytorch_lightning.loggers.WandbLogger` is available anywhere except
...
+    # Display a single run
+    %wandb USERNAME/PROJECT/runs/RUN_ID
+    # Display a sweep
+    %wandb USERNAME/PROJECT/sweeps/SWEEP_ID
+    # Display a report
+    %wandb USERNAME/PROJECT/reports/REPORT_ID
+    # Specify the height of embedded iframe
+    %wandb USERNAME/PROJECT -h 2048
+
+More information is available `here https://docs.wandb.ai/guides/track/jupyter`__.
```
Comment (rohitgr7) [inline docs/source/common/loggers.rst:256]:
```suggestion
More information is available `here <https://docs.wandb.ai/guides/track/jupyter>`__.
```

**Review by kaushikb11 — APPROVED** [review]

**Pull request merged**